Rakefile

```diff
@@ -25,16 +25,24 @@ namespace :test do
   task templates: "templates:all"
 
   namespace :templates do
-    task all: [ :daemonize, :npm, :rackup, :mocha, :kill, :exit ]
+    task all: [ :daemonize, :npm, :rackup, :wait, :mocha, :kill, :exit ]
     task serve: [ :npm, :rackup ]
 
     work_dir    = Pathname(__FILE__).dirname.join("test/templates")
     pid_file    = Pathname(Dir.tmpdir).join("web_console.#{SecureRandom.uuid}.pid")
-    server_port = 29292
-    rackup_opts = "-p #{server_port}"
-    test_runner = "http://localhost:#{server_port}/html/spec_runner.html"
+    runner_uri  = URI.parse("http://localhost:29292/html/spec_runner.html")
+    rackup_opts = "-p #{runner_uri.port}"
     test_result = nil
 
+    def need_to_wait?(uri)
+      begin
```

Comment:
No need for the `begin` here. The body of the method acts as implicit begin block, so you can do:

```ruby
def some_method
  call_an_action_that_may_raise_an_error
rescue SomeErrorType
  handle_it_here
end
```